src/app/topic/topic.component.ts

```diff
@@ -90,6 +93,18 @@ export class TopicComponent extends RouterEnter {
     }
     this._getTopic();
     this._getTopicProgress();
+    // convert other brand video players to custom player.
+    setTimeout(() => {
+      Array.from(this.document.querySelectorAll('#video')).map(player => {
```

Comment:
we only have one `#video`, don't need to do a queryselectall

Reply:
yes in our component we have only one element with id="video", but if you check story i have mention in there this also working with embedded videos. so we can't say only one video embedded. if video added from video URL html element in the code will show it but if video is embedded that html code shows after view loaded.

Reply:
because we have embedded videos and also video URL that's why i use `queryselectall`. if their only one element or multiple element `queryselectall` can handle that.

Reply:
i have test in practera web, we can create topics that have multiple embedded videos. and also we can create topic that have both embedded video and video URL.

Reply:
How do we make sure the embedded video has the same id `video` as in our component?

Reply:
If we have to deal with the embedded video, shall we consider using a smarter way to detect those videos? 
We can't guarantee that all videos have the `id="video"`. 
If we are dealing with only `<iframe>` and  `<video>` these two scenarios, we can use the CSS query selector to get those DOMs, and make sure we are only picking those DOMs inside the `<app-topic>`

Reply:
we need to add that when we create topic. i had chat with anu about this. we need to inform this to programs team. when they embedded videos they need to add an extra `div` with `id` and `class`. i have include examples in the jira story. if they didn't add this extra `div` video will play on other branded players. what anu suggested is. let's try this first if programs team have difficulties to add this extra html code when embedding video we may need to change that.

Reply:
Also, if we have to do a `querySelectorAll`, it's better to do `querySelectorAll` against a class name instead of id. Something like `querySelectorAll('.topic-video')`. `id` of an HTML element is meant to be unique, while `class` is used to set the element under a "group".

Reply:
Also, we are not actually "mapping" an Array, we are just looping through it, so instead of doing an `Array.map()`, use `Array.forEach()` instead

Reply:
yes we can use CSS query selector.

Reply:
ok will change it to Array.forEach().